packages/bundlers/default/src/DefaultBundler.js

```diff
@@ -157,11 +178,23 @@ function decorateLegacyGraph(
           invariant(dependency.type === 'dependency');
           return dependency.value;
         });
+      invariant(
+        entryAsset != null,
+        'Processing a bundleGroup with no entry asset',
+      );
       for (let dependency of dependencies) {
-        bundleGroup = bundleGraph.createBundleGroup(
-          dependency,
-          idealBundle.target,
-        );
+        if (
+          nullthrows(bundleGraph.getResolvedAsset(dependency)).id ===
+          entryAsset.id
+        ) {
+          bundleGroup = bundleGraph.createBundleGroup(
+            dependency,
+            idealBundle.target,
+          );
+          bundleGroups.push(bundleGroup);
+        } else {
+          //Warning?
```

Comment:
Can/should this situation actually occur? If not then maybe we should just assert?

Reply:
I believe this is leftover from when the implementation allowed for MSBs to have a main entry. I think this snippet executed if an MSB was also a BundleGroup, and then some of the dependencies resolved to a non-entry asset. 

I think we could probably remove the logic and maybe invariant that no bundleGroups are MSBs since that should never be the case... I can do that.

Reply:
Done, also tested internally 👍